collection requirements: Add a note about CI jobs limits (#40)
ansible/ansible-documentation: docs/docsite/rst/community/collection_contributors/collection_requirements.rst

  You can copy the free-to-use `GitHub action workflow file <https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml>`_ from the `Collection Template repository <https://github.com/ansible-collections/collection_template/>`_ to the `.github/workflows` directory in your collection to set up testing through GitHub actions. The workflow covers all the requirements below.

.. note::

  Subscribe to the `news-for-maintainers repository <https://github.com/ansible-collections/news-for-maintainers>`_
  and keep your matrix up to date in accordance to related announcements.

>   Timely add new ansible-core versions and consider dropping support and testing against its EOL versions.

Reviewer: ```suggestion
  Add new `ansible-core` versions in a timely manner and consider dropping support and testing against its EOL versions.
```

Other requirements
===================

> * After content is moved out of another currently included collection such as ``community.general`` or ``community.network`` OR a new collection satisfies all the requirements, add the collection to the ``ansible.in`` file in a corresponding directory of the `ansible-build-data repository <https://github.com/ansible-community/ansible-build-data/>`_.

Reviewer 2: ```suggestion
* After content is moved out of another currently included collection such as ``community.general`` or ``community.network`` OR a new collection satisfies all the requirements, see `Adding a new collection <https://github.com/ansible-community/ansible-build-data/#adding-a-new-collection>`_ in the `ansible-build-data repository <https://github.com/ansible-community/ansible-build-data/>`_'s README.
```

Author: @felixfontein merged, thanks! PTAL
